### PR TITLE
chore: update root /templates to use `wrangler generate` instead of `create-cloudflare@v1`

### PR DESCRIPTION
**Pull request by admah**
Fixes # [insert GH or internal issue number(s)].

**What this PR solves / how to test:**

This updates all template READMEs to include `--no-delegate-c3` any time that the `init` command is used. This allows for each template to be initialized as expected in a project folder. 

There are also a number of docs fixes in certain templates like: proper heading structure, adding Wrangler sections, and fixing other lint-type errors.

**Associated docs issue(s)/PR(s):**

- [insert associated docs issue(s)/PR(s)]

**Author has included the following, where applicable:**

- [ ] Tests
- [ ] Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))

**Reviewer is to perform the following, as applicable:**

- Checked for inclusion of relevant tests
- Checked for inclusion of a relevant changeset
- Checked for creation of associated docs updates
- Manually pulled down the changes and spot-tested

### CODE REVIEW

File: templates/experimental/worker-dart/README.md
Comment (petebacondarwin) [inline templates/experimental/worker-dart/README.md:19]: This is not accurate any more, right? Since the `--no-delegate-c3` actually does not invoke create-cloudflare. And similarly throughout these templates.

**Review by petebacondarwin — COMMENTED** [review]
Also the following syntax would work, which I think actually read a bit better:

```
--delegate-c3=false
```

**Comment by Cherry** [comment]
Reported this to Adam privately, but it's worth noting before merge that `--no-delegate-c3` and `--delegate-c3=false` don't seem to actually work right now:



Wouldn't the commands have to be changed to use `npx wrangler@latest init ... --no-delegate-c3` to get the `wrangler init` behaviour? But `init` doesn't support templates like this, does it?

The only way I've been able to get these templates to work in wrangler 3 (without using older `create-cloudflare versions), is via `npx wrangler@latest generate my-project worker-cobol`.

File: templates/experimental/worker-dart/README.md
Comment (Cherry) [reply templates/experimental/worker-dart/README.md:19]: Isn't this only ever going to invoke `create-cloudflare` though, since that's what `npm init` does. It's not in `wrangler init`, so there's nothing it can't "not delegate" to - it's only using C3?

`delegate-c3` is a `wrangler` option and not a `create-cloudflare` option, right?

Comment (admah) [reply templates/experimental/worker-dart/README.md:19]: That's correct. I'm wondering if the  best solution for now might be to run `npm create cloudflare@1 [project-name] [template-name]` since that will essentially copy the template and create the directory as expected.

Comment (petebacondarwin) [reply templates/experimental/worker-dart/README.md:19]: LOL! Sorry I only just realised that it was saying `npm init cloudflare` rather than `npx wrangler init`. 

Good catch @Cherry!

Yeah, I guess just reverting these docs to point at create-cloudflare@1 for now is the simplest solution.

**Comment by Cherry** [comment]
If it makes more sense to use `wrangler generate` here, I'm happy to PR that if needed. We still get people running into this problem regularly.

**Comment by admah** [comment]
@Cherry that does seem to work better for now. I've updated the templates and the README to reflect that.

**Comment by Cherry** [comment]
Any chance we can get this merged in? 👀 People are still hitting this regularly.

**Review by Skye-31 — APPROVED** [review]
LGTM pending prettier fix

**Review by RamIdeas — APPROVED** [review]

**Pull request merged**
